src/dra/menu.c

```diff
@@ -3020,8 +3020,214 @@ bool func_800FB1EC(s32 arg0) {
+    s32 var_s6;
+    u16 otherItemId;
+    u16 yetAnotherId;
+
+    s32 temp_psp_a1;
+    s32 temp_psp_a2;
+
+    nItems = func_800FD6C4(D_801375CC);
+    prevCursor = nav->cursorMain;
+    MenuHandleCursorInput(nav, nItems, 2);
+    itemId = order[D_801375D8[nav->cursorMain]];
+    otherItemId = *selected;
+    ret = 0;
+    if (count[itemId] != 0) {
+        *selected = itemId;
+    } else {
+        *selected = 0;
+    }
+
+    if (D_801375CC == HAND_TYPE) {
+        yetAnotherId = g_Status.equipment[1 - D_801375D0];
+        if (count[itemId] > 0) {
+            if (g_EquipDefs[otherItemId].itemCategory == 5) {
+                g_Status.equipment[1 - D_801375D0] = 0;
+            }
+            if (g_EquipDefs[itemId].itemCategory == 5) {
+                g_Status.equipment[1 - D_801375D0] = itemId;
+            }
+        } else if (g_EquipDefs[otherItemId].itemCategory == 5) {
+            g_Status.equipment[1 - D_801375D0] = 0;
+        }
+    }
+    var_s6 = 0;
+    func_800F53A4();
+
+    if ((g_Player.unk0C & 0x17) | (PLAYER.step == Player_UnmorphWolf) |
+        (PLAYER.step == Player_BossGrab) | (g_Player.unk60)) {
+        if (itemId == ITEM_AXE_LORD_ARMOR) {
+            if (D_801375CC == ARMOR_TYPE) {
+                if (count[ITEM_AXE_LORD_ARMOR] != 0) {
+                    var_s6 = 1;
+                }
+            }
+        }
+    }
+
+    temp_psp_a2 = 0;
+    if (D_801375CC == HAND_TYPE && count[itemId] > 0) {
+        if (itemId == ITEM_LIFE_APPLE || itemId == ITEM_HAMMER) {
+            temp_psp_a1 = 1;
+        } else {
+            temp_psp_a1 = 0;
+        }
+        if (temp_psp_a1) {
+            temp_psp_a2 = 1;
+        }
+    }
+    var_s6 |= temp_psp_a2;
+    D_80137948 = 0;
+    if (var_s6 == 0) {
+        D_80137948 = 1;
+        func_800F7244();
+    }
+    *selected = otherItemId;
+    if (D_801375CC == HAND_TYPE) {
+        g_Status.equipment[1 - D_801375D0] = yetAnotherId;
+    }
+    func_800F53A4();
+    if (g_pads[0].tapped & PAD_MENU_SORT) {
+        if (g_IsSelectingEquipment == 0) {
+            if (func_800FB1EC(itemId) == false) {
+                g_EquipmentCursor = nav->cursorMain;
+                g_IsSelectingEquipment++;
+                PlaySfx(0x633);
+            } else {
+                PlaySfx(0x686);
+            }
+        } else if (func_800FB1EC(itemId) != false) {
+            PlaySfx(0x686);
+        } else {
+            goto block_36;
+        }
+    } else if (g_pads[0].tapped & PAD_MENU_SELECT) {
+        if (g_IsSelectingEquipment != 0) {
+            do {
+                if (func_800FB1EC(itemId) == false) {
+                block_36:
+                    func_800FB160(
+                        g_EquipmentCursor, nav->cursorMain, D_801375CC);
+                    ret = 2;
+                    g_IsSelectingEquipment = 0;
+                    PlaySfx(0x633);
+                } else {
+                    PlaySfx(0x686);
+                }
+            } while (0);
```

Comment:
Does the `do/while(0)` generate different code than without it?

Reply:
Yes, unfortunately this is a common hack that ends up making code match, which otherwise does not match. In this case, see the scratch: https://decomp.me/scratch/BghLP

If you remove the do-while, you get the `s5` and `s6` registers swapped throughout the function.

If you can find an alternative fix, that would be very appreciated.